apps/docs/docs/cdk/transformations/transformations.mdx

```diff
@@ -8,19 +8,46 @@ hide_title: true
 # @rx-angular/cdk/transformations
 
 [![npm](https://img.shields.io/npm/v/%40rx-angular%2Fcdk.svg)](https://www.npmjs.com/package/%40rx-angular%2Fcdk)
-![rx-angular CI](https://github.com/rx-angular/rx-angular/workflows/rx-angular%20CI/badge.svg?branch=master)
+![rx-angular CI](https://github.com/rx-angular/rx-angular/workflows/rx-angular%20CI/badge.svg?branch=main)
 
 > A set of functions to manipulate the state in an efficient way.
 
-`@rx-angular/cdk/transformations` provides a set of functions to perform state transformations in a performant way.
+- `@rx-angular/cdk/transformations` provides a set of tools that simplifies the management of data structures in an immutable way.
 
-## Key features
+Currently transformation helpers support the management of objects and arrays.
 
-- ✅
+You can use the transformation helpers without RxState but expect the behavior to be opinionated since in the first place this package is designed to optimize state management with RxState.
 
-## Demos:
+## Principles
 
-- ⚡ GitHub
+**Helpers that initialize or modify state property**
+
+```
+Array: insert, update,
+Object: patch, setProp, toggle
+```
+
+If original property not matching expected type (expected array, got string) or original property is null/undefined. Provided updates will be returned.
+
+**Helpers that should remove something from non-primitive state property or remove state property**
+
+```
+Array: remove
+Object: deleteProp
+```
+
+If the original property not matching the expected type (expected array, got string) an empty array or object will be returned.
+If the original property is null or undefined helper will return null or undefined.
+
+**Helpers that converting one structure to another**
+
+```
+Array: dictionaryToArray
+Object: toDictionary
+```
+
+If the original structure not matching the expected type empty target structure will be returned.
+If the original structure is null or undefined helper will return null or undefined.
```

Comment:
What about adding more general information, but not implementation details (like what it returns) here:

- Manage objects and collections with an immutable guarantee
- Reduce boilerplate to perform state changes
- Type-safe and tree-shakeable functions
- Ensure serializable objects and arrays

What do you think?

Reply:
nice, I can put it as `Key features`

like here https://www.rx-angular.io/docs/cdk/coalescing#key-features

Reply:
added `Key features` list instead of `Principles`